include/seastar/json/formatter.hh

```diff
@@ -191,23 +220,19 @@ public:
     static sstring to_json(bool d);
 
     /**
-     * return a json formatted list of a given vector of params
-     * @param vec the vector to format
-     * @return the given vector in a json format
+     * converts a given range to a JSON-formatted string
+     *
+     * @param range A standard range type
+     * @return A string containing the JSON representation of the input range
      */
-    template<typename... Args>
-    static sstring to_json(const std::vector<Args...>& vec) {
-        return to_json(state::array, vec.begin(), vec.end());
-    }
-
-    template<typename... Args>
-    static sstring to_json(const std::map<Args...>& map) {
-        return to_json(state::map, map.begin(), map.end());
-    }
-
-    template<typename... Args>
-    static sstring to_json(const std::unordered_map<Args...>& map) {
-        return to_json(state::map, map.begin(), map.end());
+    template<std::ranges::input_range Range>
+    requires (!internal::is_string_like<Range>)
+    static sstring to_json(const Range& range) {
```

Comment:
You'll want to do the same for `formatter::write` such that the zero copy methods get the same support.

Reply:
@StephanDollberg hi Stephan, thank you for your review. added the support to `formatter::write()` in the latest revision.